src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlConnectionHelper.cs

```diff
@@ -86,8 +86,10 @@ private void ConnectionString_Set(DbConnectionPoolKey key)
             {
                 throw ADP.OpenConnectionPropertySet(nameof(ConnectionString), connectionInternal.State);
             }
-            string cstr = ((null != connectionOptions) ? connectionOptions.UsersConnectionStringForTrace() : "");
-            SqlClientEventSource.Log.TraceEvent("<prov.DbConnectionHelper.ConnectionString_Set|API> {0}, '{1}'", ObjectID, cstr);
+            if (SqlClientEventSource.Log.IsTraceEnabled())
+            {
+                SqlClientEventSource.Log.TraceEvent("<prov.DbConnectionHelper.ConnectionString_Set|API> {0}, '{1}'", ObjectID, (null != connectionOptions) ? connectionOptions.UsersConnectionStringForTrace() : "");
+            }
```

Comment:
I'm not sure this is a good pattern but If you want to keep it, remove the redundant codes inside the function `TraceEvent` and other functions in EventSource.

Reply:
Good point, this pattern will improve the performance when the eventsource is not requested, but double check will drop the perfomance when eventsource log is requested. The solution would be to add a new NonEvent function to EventSource without the if checker and call these functions from there. do you agree?